lib/pages/timer/timer_logic.dart

```diff
@@ -67,117 +62,125 @@ class TimerLogic extends ChangeNotifier {
     }
   }
 
-  Future<void> saveTimesToPrefs() async {
-    final prefs = await SharedPreferences.getInstance();
-    // Save finished times
-    prefs.setString('finishedWorkTimes',
-        jsonEncode(finishedWorkTimes.map((d) => d.inMilliseconds).toList()));
-    prefs.setString('finishedPauseTimes',
-        jsonEncode(finishedPauseTimes.map((d) => d.inMilliseconds).toList()));
-    prefs.setString('finishedDrivingTimes',
-        jsonEncode(finishedDrivingTimes.map((d) => d.inMilliseconds).toList()));
-
-    // Save start times
-    prefs.setString(
-        'workTimeStartTime', workTimeStartTime?.toIso8601String() ?? '');
-    prefs.setString('pauseStartTime', pauseStartTime?.toIso8601String() ?? '');
-    prefs.setString(
-        'drivingTimeStartTime', drivingTimeStartTime?.toIso8601String() ?? '');
-
-    // Save today's date
-    prefs.setString(
-        'savedDate', DateFormat('yyyy-MM-dd').format(DateTime.now()));
-  }
-
-  Future<void> loadTimesFromPrefs() async {
-    final prefs = await SharedPreferences.getInstance();
-    String todayDate = DateFormat('yyyy-MM-dd').format(DateTime.now());
-    String? savedDate = prefs.getString('savedDate');
-
-    if (savedDate != null && savedDate == todayDate) {
-      // Load finished times
-      String? finishedWorkTimesJson = prefs.getString('finishedWorkTimes');
-      if (finishedWorkTimesJson != null && finishedWorkTimesJson.isNotEmpty) {
-        List<dynamic> millisList = jsonDecode(finishedWorkTimesJson);
-        finishedWorkTimes =
-            millisList.map((ms) => Duration(milliseconds: ms)).toList();
-      }
-
-      String? finishedPauseTimesJson = prefs.getString('finishedPauseTimes');
-      if (finishedPauseTimesJson != null && finishedPauseTimesJson.isNotEmpty) {
-        List<dynamic> millisList = jsonDecode(finishedPauseTimesJson);
-        finishedPauseTimes =
-            millisList.map((ms) => Duration(milliseconds: ms)).toList();
-      }
-
-      String? finishedDrivingTimesJson =
-          prefs.getString('finishedDrivingTimes');
-      if (finishedDrivingTimesJson != null &&
-          finishedDrivingTimesJson.isNotEmpty) {
-        List<dynamic> millisList = jsonDecode(finishedDrivingTimesJson);
-        finishedDrivingTimes =
-            millisList.map((ms) => Duration(milliseconds: ms)).toList();
-      }
-
-      // Load start times
-      String? workTimeStartTimeString = prefs.getString('workTimeStartTime');
-      if (workTimeStartTimeString != null &&
-          workTimeStartTimeString.isNotEmpty) {
-        workTimeStartTime = DateTime.parse(workTimeStartTimeString);
+  Future<void> fetchTimersForToday() async {
+    const String query = r'''
+      query TimersInBoundary($lowerBound: DateTime!, $upperBound: DateTime!) {
```

Comment:
Minor improvment. There is a TimersToday (and TimersCurrentMonth) function, that calculates the lower- and upper- Bounds for you.

Reply:
[The Line in the Backend](https://github.com/Squad-Mandalore/ttbackend/blob/7a4d372587d150b29a86faff40beb0b895b29821/src/graphql/timer.rs#L34)